python/lsst/daf/base/baseLib.i

```diff
@@ -161,6 +161,48 @@ def _propertyContainerGet(container, name, asArray=False):
         pass
     raise lsst.pex.exceptions.TypeError('Unknown PropertySet value type for ' + name)
 
+import numbers
+def _guessIntegerType(container, name, value):
+    """Given an existing container and name, determine the type
+    that should be used for the supplied value. The supplied value
+    is assumed to be a scalar.
+
+    On Python 3 all ints are LongLong but we need to be able to store them
+    in Int containers if that is what is being used (testing for truncation).
+    Int is assumed to mean 32bit integer (2147483647 to -2147483648).
+
+    If there is no pre-existing value we have to decide what to do. For now
+    we pick Int if the value is less than maxsize.
+
+    Returns None if the value supplied is a bool or not an integral value.
+    """
+    useType = None
+    maxInt = 2147483647
+    minInt = -2147483648
+
+    # We do not want to convert bool to int so let the system work that
+    # out itself
```

Comment:
You had to wait for afw to fail before you knew there was a problem here?  Can you add a test that will catch the problem in this package?